### PR TITLE
documentation for Incremental build

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

File: Documentation/incrementalbuild/intro_incremental_build.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+Before `Compile` phase, only **changed files and their dependencies** would be loaded. During `Compile` phase, rebuilt articles could report new dependencies. After `Compile` phase, newly introduced dependencies would lead to some unchanged files being reloaded.
+
+*DocFX Incremental Build Framework* provides [a flexible way](customize_a_processor_to_support_incremental.md) to define plugins/processors to support incremental build. Plugins that support incremental are free to [report dependencies](advanced_report_dependency.md) at `Compile` phase.
+
+> [!Note]
+> Only `Compile` phase could report/collect dependencies.
+
+Framework also allows plugins to save/load context related info in Plugin Cache. Details please refer to [Plugin cache](customize_a_processor_to_support_incremental.md#step3-optionalimplement-microsoftdocascodepluginsicantracecontextinfobuildstep-interface-for-plugins-that-need-to-access-context-info).
```
Comment (superyyrrzz) [inline Documentation/incrementalbuild/intro_incremental_build.md]: >Details [](start = 81, length = 7)

please refer to ... for more detail

File: Documentation/incrementalbuild/intro_incremental_build.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+By default, incremental cache files will be put at path `obj/.cache/build/` relative to your `docfx.json`. You're also free to specify the path with option `--intermediateFolder`. About the structure of the cache folder please refer to [Cache file structure](advanced_cache_file_structure.md).
+
+
+*Incremental Condition*
+------------------------
+Build could run incrementally only if all of the following conditions meet.
+
+1. The version supports incremental.
```
Comment (superyyrrzz) [inline Documentation/incrementalbuild/intro_incremental_build.md]: >version [](start = 7, length = 7)

Add some explanation for version?

File: Documentation/incrementalbuild/advanced_report_dependency.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+DocFX Incremental Build Framework provides the flexiblity to register customized dependency type and report dependency. In this topic, we will go through how to do that.
+
+*Register Dependency Type*
+---------------------------
+
+### Dependency type
+
+Property                 | Type                                 | Description
+-----------------------  | ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------
+**Name**                 | string                               | name of the dependency type(it should be unique)
```
Comment (superyyrrzz) [inline Documentation/incrementalbuild/advanced_report_dependency.md]: >**Name** [](start = 0, length = 8)

unbold Name?

File: Documentation/incrementalbuild/advanced_report_dependency.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+If you want to report the dependency between an file and another file, you can use below method:
+
+directed dependency: @Microsoft.DocAsCode.Plugins.IHostService.ReportDependencyTo(Microsoft.DocAsCode.Plugins.FileModel,System.String,System.String)
+
+reversed dependency: @Microsoft.DocAsCode.Plugins.IHostService.ReportDependencyFrom(Microsoft.DocAsCode.Plugins.FileModel,System.String,System.String)
+
+For example, i'd like to report a dependency: file `~/test.md`(filemodel is `a`) depends on file `~/../include/token/md`, dependency type is `include`,
+i could call the method `ReportDependencyTo(a, "~/../include/token.md", "include")`.
+
+Plugins are only allowed to report a dependency during `Compile` phase. However, some plugins might don't have enough info to resolve some dependency to file until the whole phase completes. Incremental build framework provides the flexiblity that plugins could report dependency between items that are not files and resolve them later.
```
Comment (superyyrrzz) [inline Documentation/incrementalbuild/advanced_report_dependency.md]: >between items [](start = 280, length = 13)

between files and items?

File: Documentation/incrementalbuild/advanced_cache_file_structure.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+In the root folder of the cache files, there is a `build.info`, it is kind of index page, which is the entry point of the cache files. Below table lists the major information inside.
+
+### BuildInfo model
+
+Property              | Type                    | Description
+--------------------- | ---------------------   | -----------------------------------------------------------
+DirectoryName         | string                  | Base directory of the cache files of last successful build
+DocfxVersion          | string                  | DocFX version
+PluginHash            | string                  | The hash of plugins plugined in DocFX
```
Comment (superyyrrzz) [inline Documentation/incrementalbuild/advanced_cache_file_structure.md]: >plugined [](start = 70, length = 8)

plugged?

File: Documentation/incrementalbuild/customize_a_processor_to_support_incremental.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
+Walkthrough: Customize a processor to support incremental build
+================================================================
+
+During this tutorial, we'll walk through the steps to enable a processor incremental.
```
Comment (hellosnow) [inline Documentation/incrementalbuild/customize_a_processor_to_support_incremental.md]: >a processor incremental [](start = 61, length = 23)

a processor to be incremental?

File: Documentation/incrementalbuild/advanced_report_dependency.md (hunk 3 above)
Comment (ansyral) [inline Documentation/incrementalbuild/advanced_report_dependency.md]: it should be unique

File: Documentation/incrementalbuild/intro_incremental_build.md (hunk 2 above)
Comment (ansyral) [inline Documentation/incrementalbuild/intro_incremental_build.md]: nice

File: Documentation/incrementalbuild/advanced_report_dependency.md (hunk 4 above)
Comment (ansyral) [inline Documentation/incrementalbuild/advanced_report_dependency.md]: we support dependency between item and item

File: Documentation/incrementalbuild/intro_incremental_build.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+Before `Compile` phase, only **changed files and their dependencies** would be loaded. During `Compile` phase, rebuilt articles could report new dependencies. After `Compile` phase, newly introduced dependencies would lead to some unchanged files being reloaded.
+
+*DocFX Incremental Build Framework* provides [a flexible way](customize_a_processor_to_support_incremental.md) to define plugins/processors to support incremental build. Plugins that support incremental are free to [report dependencies](advanced_report_dependency.md) at `Compile` phase.
+
+> [!Note]
+> Only `Compile` phase could report/collect dependencies.
+
+Framework also allows plugins to save/load context related info in Plugin Cache. Details please refer to [Plugin cache](customize_a_processor_to_support_incremental.md#step3-optionalimplement-microsoftdocascodepluginsicantracecontextinfobuildstep-interface-for-plugins-that-need-to-access-context-info) for more details.
```
Comment (superyyrrzz) [inline Documentation/incrementalbuild/intro_incremental_build.md]: >Details [](start = 81, length = 7)

remote

**Comment by superyyrrzz** [comment]
:shipit:

File: Documentation/docfx.json [hunk 8]
```diff
@@ -23,7 +23,7 @@
         "src": ".."
       },
       {
-        "files": [ "tutorial/**/*.{md,yml}", "spec/**/*.{md,yml}", "*.{md,yml}", "guideline/**/*.{md,yml}", "templates-and-plugins/**/*.{md,yml}" ]
+        "files": [ "tutorial/**/*.{md,yml}", "spec/**/*.{md,yml}", "*.{md,yml}", "guideline/**/*.{md,yml}", "templates-and-plugins/**/*.{md,yml}", "incrementalbuild/**/*.{md,yml}" ]
```
Comment (vicancy) [inline Documentation/docfx.json]: >incrementalbuild [](start = 148, length = 16)

move into spec?

File: Documentation/toc.yml [hunk 9]
```diff
@@ -7,3 +7,5 @@
   href: obj/temp/docfxapi/
 - name: "Templates & Plugins"
   href: templates-and-plugins/
+- name: Incremental Build
```
Comment (vicancy) [inline Documentation/toc.yml]: >Incremental Build [](start = 8, length = 17)

should this be a sub node for spec?

File: Documentation/docfx.json (hunk 8 above)
Comment (ansyral) [inline Documentation/docfx.json]: how about tutorial since user might need to reference it to customize their processor?

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**